File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/ApplePie.java
package content.global.skill.production.cooking.recipe.pie.impl;

import content.global.skill.production.cooking.recipe.pie.PieRecipe;
import core.game.node.item.Item;

/**
 * The Apple pie.
 */
public class ApplePie extends PieRecipe {

    private static final Item UNCOOKED_PIE = new Item(2317);

    private static final Item COOKING_APPLE = new Item(1955);

    @Override
    public Item getProduct() {
        return UNCOOKED_PIE;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{COOKING_APPLE};
    }

}

File: Emulator/src/main/content/global/skill/production/cooking/recipe/pie/impl/ApplePie.kt
package content.global.skill.production.cooking.recipe.pie.impl

import content.global.skill.production.cooking.recipe.pie.PieRecipe
import org.rs.consts.Items
import core.game.node.item.Item

/**
 * Apple pie.
 */
class ApplePie : PieRecipe() {

    override fun getProduct(): Item {
        return UNCOOKED_PIE
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(COOKING_APPLE)
    }

    companion object {
        private val UNCOOKED_PIE = Item(Items.UNCOOKED_APPLE_PIE_2317)
        private val COOKING_APPLE = Item(Items.COOKING_APPLE_1955)
    }
}
